eth/transactionManager.go

```diff
@@ -0,0 +1,241 @@
+package eth
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"strings"
+	"sync"
+	"time"
+
+	ethereum "github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/event"
+	"github.com/golang/glog"
+	"github.com/livepeer/go-livepeer/common"
+)
+
+type TransactionManager struct {
+	txTimeout       time.Duration
+	maxReplaceMents int
+
+	queue transactionQueue
+
+	// subscriptions
+	feed  event.Feed
+	scope event.SubscriptionScope
+
+	eth *ethclient.Client
+	gpm *GasPriceMonitor
+	am  AccountManager
+
+	cond sync.Cond
+
+	quit chan struct{}
+}
+
+type transactionQueue []*types.Transaction
+
+type transactionReceipt struct {
```

Comment:
I went with `transactionReceipt` since the receipt values are embedded in this type (through albeit through a pointer) with the addition of an error (which can be the revert message in some cases).